hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/ws/server/AbstractGatewayWebSocket.java

```diff
@@ -0,0 +1,221 @@
+        String password = null;
+        String token = null;
+        String persona = null;
+
+        if (basicMessage instanceof AuthMessage) {
+            AuthMessage authMessage = (AuthMessage) basicMessage;
+            Authentication auth = authMessage.getAuthentication();
+            if (auth != null) {
+                username = auth.getUsername();
+                password = auth.getPassword();
+                token = auth.getToken();
+                persona = auth.getPersona();
+
+                // make sure we clear the auth message - if we forward it we don't want the creds going along with it
+                authMessage.setAuthentication(null);
+            }
+        }
+
+        try {
+            // note that if both username and token are provided, we authenticate using the token
+            boolean hasToken = (token != null && !token.isEmpty());
+            if (hasToken) {
+                log.tracef("authenticating token [%s/%s], session=[%s]", token, persona, session.getId());
+                authenticator.authenticateWithToken(token, persona, session);
+            } else {
+                log.tracef("authenticating user [%s/%s/%s], session=[%s]", username, password, persona,
+                        session.getId());
+                authenticator.authenticateWithCredentials(username, password, persona, session);
+            }
+        } catch (WebsocketAuthenticationException wae) {
+            throw wae;
+        } catch (Exception e) {
+            throw new WebsocketAuthenticationException("Unauthorized!", e);
+        }
+
+        return; // authentication successful
+    }
+
+    protected void handleRequest(Session session, BasicMessageWithExtraData<BasicMessage> requestWithBinary)
+            throws WebsocketAuthenticationException, NoCommandForMessageException, Exception {
+        BasicMessage request = requestWithBinary.getBasicMessage();
+
+        if (request instanceof UiSessionOrigin) {
+            /* do not trust the sessionId provided by the client */
+            log.tracef("[%s] is an instance of [%s]", request.getClass().getName(),
+                    UiSessionOrigin.class.getName());
+            ((UiSessionOrigin) request).setSenderSessionId(session.getId());
+        }
+
+        // make sure the user is authenticated
+        authenticate(request, session);
+
+        @SuppressWarnings("unchecked")
+        WsCommand<BasicMessage> command = (WsCommand<BasicMessage>) wsCommands.getCommand(request.getClass());
+        log.debugf("About to execute command [%s] on message [%s] in session [%s] of [%s]", command.getClass(),
+                request.getClass().getName(), session.getId(), endpoint);
+        WsCommandContext context = commandContextFactory.newCommandContext(session);
+        command.execute(requestWithBinary, context);
+    }
+
+    /**
+     * When a binary message is received from a UI client, this method will execute the command the client is asking
```

Comment:
This mentions "UI Client" but this class is for both feeds and UI clients, so just make the javadocs more general. Same thing in the @return description